app/main.py

```diff
@@ -1,13 +1,47 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+
+    return {
+        "errors":
+            [
+                {
+                    "line": err_single_file["line_number"],
+                    "column": err_single_file["column_number"],
+                    "message": err_single_file["text"],
+                    "name": err_single_file["code"],
+                    "source": "flake8"
+                } for err_single_file in errors
```

Comment:
The logic for formatting errors in `format_single_linter_file` is similar to the one in `format_linter_report`. Consider refactoring this logic into a separate function to avoid redundancy and improve maintainability.